client/hwlib/src/models/request_validators.rs

```diff
@@ -48,17 +48,27 @@ pub struct Paths {
     pub cpuinfo_filepath: PathBuf,
     pub max_cpu_frequency_filepath: PathBuf,
     pub device_tree_dirpath: PathBuf,
+    pub os_release_filepath: PathBuf,
     pub proc_version_filepath: PathBuf,
 }
 
 impl Default for Paths {
     fn default() -> Self {
+        // Snap confinement exposes /etc/os-release in a different location
+        let snap_os_release_filepath = PathBuf::from(constants::SNAP_OS_RELEASE_FILE_PATH);
+        let os_release_filepath = if snap_os_release_filepath.exists() {
+            snap_os_release_filepath
+        } else {
+            PathBuf::from(constants::OS_RELEASE_FILE_PATH)
+        };
```

Comment:
This block looks surprising to me, do we need `hwlib` to be aware of two possible paths depending on how it's packaged? This feels very fragile to me, someone coming to this code in future may easily forget to check the values in this order, quietly leading to bugs. Instead, would be be cleaner to just have one `RELEASE_FILE_PATH` which is configured correctly at compile time?